tests/unit/config/config_test.py

```diff
@@ -5166,6 +5166,28 @@ def test_serialize_escape_dollar_sign(self):
         assert serialized_service['command'] == 'echo $$FOO'
         assert serialized_service['entrypoint'][0] == '$$SHELL'
 
+    def test_serialize_escape_dont_interpolate(self):
```

Comment:
Can we also add a test that verifies section validation is still enforced with `interpolate=False`? Basically a copy of `test_config_integer_service_name_raise_validation_error_v2` (line 598) but with `interpolate=False` in the `config.load` call?